spinalcordtoolbox/reports/slice.py

```diff
@@ -57,8 +57,15 @@ def __init__(self, images, p_resample=0.6):
             self._absolute_paths.append(img.absolutepath)  # change_orientation removes the field absolute_path
             img.change_orientation('SAL')
             if p_resample:
-                if i == len(images) - 1 and img.dim[3] == 1:
-                    # Last volume, if it is 3d, corresponds to a segmentation, therefore use linear interpolation here
+                # Check if image is a segmentation (binary or soft) by making sure:
+                # - 0/1 are the two most common voxel values
+                # - 0/1 account for >99% of voxels (to allow for some soft voxels)
```

Comment:
I guess this works for now, but doing this kind of auto-detection is always fragile. Really, the calling code should _know_ which images are intended to be segmentations without guessing, and the information shouldn't get lost along the control flow. But I understand that this is hard to do with the current code architecture, and out of scope for this PR.